### PR TITLE
Added workaround for using analog pins with binary_sensors

### PR DESCRIPTION
**Pull request by christianschneider89**
See https://github.com/home-assistant/core/issues/63371

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @christianschneider89, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#21047).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: source/_integrations/firmata.markdown [hunk 1]
```diff
@@ -142,9 +142,9 @@ binary_sensor:
       required: true
       type: string
     pin:
-      description: The digital or analog pin number on the board.
+      description: The digital or analog pin number on the board. For analog pins the corresponding digital number must be used. Please use the pinout that corresponds to your board. For example, on the Arduino Uno, pin A2 corresponds to the digital pin number 16.
       required: true
-      type: [integer, string]
+      type: integer
```
Comment (frenck) [inline source/_integrations/firmata.markdown]: This is unfortunately incorrect/not true, the integration accepts a string.

See: https://github.com/home-assistant/core/blob/5fa6ef2849a9412e97526c98187ea68ed8b204f8/homeassistant/components/firmata/__init__.py#L81

And: <https://github.com/home-assistant/core/blob/5fa6ef2849a9412e97526c98187ea68ed8b204f8/homeassistant/components/firmata/__init__.py#L50>

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/firmata.markdown (hunk 1 above)
Comment (christianschneider89) [reply source/_integrations/firmata.markdown]: Ok, I reverted this commit.

Comment (DaAwesomeP) [reply source/_integrations/firmata.markdown]: Yes, it *should* accept the form `A#` for pins (and that worked originally). This pull documents a less-intuitive workaround, not the intended method. See https://github.com/home-assistant/core/issues/63371

**Review by frenck — APPROVED** [review]
Thanks, @christianschneider89 👍

**Pull request merged**